app/lib/package/backend.dart

```diff
@@ -1269,6 +1319,8 @@ class PackageBackend {
   Future<void> removePackageTarball(String package, String version) async {
     final object = tarballObjectName(package, version);
     await deleteFromBucket(_bucket, object);
+    await deleteFromBucket(_publicBucket, object);
+    await deleteFromBucket(_canonicalBucket, object);
```

Comment:
Maybe later we should split this into 2 methods:
 * removePublicPackageTarball
 * removeCanonicalPackageTarball

Reply:
Maybe we remove only from the public bucket, and removing from the canonical is behind admin-only API?

Reply:
Yeah, we did, but it has not been started yet.